python/paddle/fluid/tests/unittests/test_program_prune_backward.py

```diff
@@ -0,0 +1,138 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+
+import contextlib
+import numpy as np
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+from test_parallel_executor_mnist import simple_fc_net, fc_with_batchnorm
+import seresnext_net
+from test_parallel_executor_transformer import transformer, get_feed_data_reader
+
+
+class TestProgramPruneBackward(unittest.TestCase):
+    def program_compare(self, program_a, program_b):
+        assert isinstance(
+            program_a, fluid.framework.
+            Program), "The first argument should be fluid.framework.Program."
+        assert isinstance(
+            program_b, fluid.framework.
+            Program), "The second argument should be fluid.framework Program."
+
+        self.assertEqual(len(program_a.blocks), len(program_b.blocks))
+        for idx in range(len(program_a.blocks)):
+            block_a = program_a.blocks[idx]
+            block_b = program_b.blocks[idx]
+            self.assertEqual(len(block_a.ops), len(block_b.ops))
+            # self.assertEqual(len(block_a.vars), len(block_b.vars))
+            for op_idx in range(len(block_a.ops)):
+                self.assertEqual(block_a.ops[op_idx].type,
+                                 block_b.ops[op_idx].type)
+            for var_key in list(block_a.vars.keys()):
+                if block_a.vars[var_key].type == core.VarDesc.VarType.RAW:
+                    continue
```

Comment:
in the `resnet`, there are three vars that are lost after `prune`.
```
name: "kCUDNNBwdFilterAlgoCache"
type {
  type: RAW
}
persistable: true

name: "kCUDNNBwdDataAlgoCache"
type {
  type: RAW
}
persistable: true

name: "kCUDNNFwdAlgoCache"
type {
  type: RAW
}
persistable: true
```
These vars are independent in scope, not associated with Op.

Reply:
these three variables are useless now, delete them in PR [#19644 ](https://github.com/PaddlePaddle/Paddle/pull/19644)